client/Fireport/Assets/Scripts/AnimationEngine/AnimationActionBehaviour.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using Game;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace AnimationEngine {
+    public abstract class AnimationActionBehaviour : MonoBehaviour {
```

Comment:
This abstract class will be the required parent of any animation action behaviours: MoveForwardAnimationBehaviour, RotateClockwiseAnimationBehaviour, etc